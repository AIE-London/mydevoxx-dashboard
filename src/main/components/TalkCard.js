/**
 * Created by dan on 04/04/2017.
 */
import React, { Component } from "react";
import Card from "./Card";
import styled from "styled-components";
import { Col, Row } from "react-flexbox-grid";

import { CommaList, CommaListItem } from "./CommaList";
import StarRating from "./StarRating";

/**
 *  Styled Components
 */
const CardHeader = styled.h2`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 2em;
  margin-bottom: 0;
`;

const CardSubhead = styled.h3`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 1.5em;
  margin-bottom: 0;
`;

const LeftPartition = styled(Col)`
  padding: 0.25em 2em;
`;

const RightPartition = styled(Col)`
  padding: 0.25em 2em;
  padding-top: 1em;
  background: #EEEEEE;
  border-left: solid 1px #BDBDBD;
  border-top: solid 1px #BDBDBD;
  text-align: left;
  padding-bottom: 1em;
`;

const CardParagraph = styled.p`
  padding-left: 1.5em;
  padding-top: 0;
  opacity: 0.6;
  text-align: left;
`;

const CardNotes = styled.textarea`
  font-size: 0.9em;
  background: rgba(255, 255, 255, 0.25);
  border: none;
  flex: 1;
  width: 100%;
  height: 10em;
  margin-top: 1em;
`;

const TrackList = styled(CommaList)`
  opacity: 0.6;
`;

const FullWidthCard = styled(Card)`
  flex: 1;
`;

const CardVideoSubhead = styled(CardSubhead)`
  color: #ff9e19;
`;

class TalkCard extends Component {
  render() {
    // [TODO] Allow the user to rate & store in indexed DB
    let rating = 0;
    let { title, summary, tracks, videoID } = this.props.talk;
    return (
      <FullWidthCard start="xs">
        <LeftPartition xs={12} md={6}>
          <Row start="xs">
            <CardHeader id="title">{title}</CardHeader>
          </Row>
          <Row end="xs">
            <StarRating rating={rating} />
          </Row>
          <CardParagraph>
            {summary}
          </CardParagraph>
          {videoID &&
            <Row>
              <CardVideoSubhead
                onClick={() => this.props.videoSelected(videoID)}
              >
                Play Video
              </CardVideoSubhead>
            </Row>}
        </LeftPartition>
        <RightPartition xs={12} md={6}>
          <Row start="xs">
            <CardSubhead id="notesHeader">My Notes</CardSubhead>
          </Row>
          <CardNotes placeholder="Write notes here..." />
          <Row>
            <CardSubhead>Top Tracks</CardSubhead>
          </Row>
          <Row>
            <TrackList>
              {tracks.map(trackName => (
                <CommaListItem key={trackName}>{trackName}</CommaListItem>
              ))}
            </TrackList>
          </Row>
        </RightPartition>
      </FullWidthCard>
    );
  }
}

export default TalkCard;
