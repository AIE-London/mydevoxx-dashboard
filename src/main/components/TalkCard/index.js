/**
 * Created by dan on 04/04/2017.
 */
import React, { Component } from 'react';
import Card from '../Card';
import styled from 'styled-components';
import {Col, Row} from 'react-flexbox-grid';

/**
 *  Styled Components
 */
const CardHeader = styled.h2`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 2em;
`;

const CardSubhead = styled.h3`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 1.5em;
  margin-bottom: 0;
`;

const LeftPartition = styled(Col)`
  padding: 0.25em 2em;
`;

const RightPartition = styled(Col)`
  padding: 0.25em 2em;
  padding-top: 1em;
  background: #EEEEEE;
  border-left: solid 1px #BDBDBD;
  text-align: left;
`;

const CardParagraph = styled.p`
  padding-left: 1.5em;
  padding-top: 0;
  opacity: 0.6;
  font-size: 0.9em;
`;

class TalkCard extends Component {
  render() {
    return (
      <Card left="xs">
        <LeftPartition xs={6}>
          <Row left="xs">
            <CardHeader id="title">{ this.props.title }</CardHeader>
          </Row>
          <p className="App-intro">
            To get started, edit <code>src/App.js</code> and save to reload.
          </p>
        </LeftPartition>
        <RightPartition xs={6}>
          <Row left="xs">
            <CardSubhead id="notesHeader">My Notes</CardSubhead>
          </Row>
          <Row left="xs">
            <CardParagraph>Lorem ipsum dolor sit amet, everti quaestio mel ea. Ex eos volutpat qualisque. Sale tantas cotidieque quo ut, ad nostro consectetuer nec. Feugiat qualisque quo an. Labores officiis te nam.</CardParagraph>
          </Row>
          <Row left="xs">
            <CardSubhead id="notesHeader">My Reviews</CardSubhead>
          </Row>
          <Row left="xs">
            <p>Feugiat qualisque quo an. Labores officiis te nam.</p>
          </Row>
        </RightPartition>
      </Card>
    );
  }
}

export default TalkCard;
