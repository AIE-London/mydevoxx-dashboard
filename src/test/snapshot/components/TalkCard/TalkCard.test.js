import React from 'react';
import ReactDOM from 'react-dom';
import TalkCard from '../../../../main/components/TalkCard';
import renderer from 'react-test-renderer';
import testImage from '../../images/test-image.jpeg';

const talkData = {
    title: 'Welcome to Devoxx',
    description: 'Join the organisers of Devoxx UK and great keynote ' +
    'speakers for inspring stories in 20 minutes.',
    rating: 3,
    topTracks: [
        'Devoxx',
        'SpringBoot',
    ],
    notes: 'Lorem ipsum dolor sit amet, everti quaestio mel ea. Ex eos ' +
    'volutpat qualisque. Sale tantas cotidieque quo ut, ad nostro consectetuer' +
    ' nec. Feugiat qualisque quo an. Labores officii.',
    review: {
        name: 'Test User 123',
        comment: '123 Great session, thanks for organising. Looking forward to the next one!',
        image: testImage
    }

};

test('card component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TalkCard title="Welcome to Devoxx"/>, div);
    });

});

test('card component with title snapshot', () => {

    const tree = renderer.create(
        <TalkCard talk={talkData}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();

});