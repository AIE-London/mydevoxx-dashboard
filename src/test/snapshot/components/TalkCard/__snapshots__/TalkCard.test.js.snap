exports[`test card component with title snapshot 1`] = `
<div
  className="kRQFnd row start-xs">
  <div
    className="col-xs-12 col-md-6 fteZLe">
    <div
      className="row start-xs">
      <h2
        className="Axdp"
        id="title">
        Welcome to Devoxx
      </h2>
    </div>
    <div
      className="row end-xs">
      <div
        className="row start-xs">
        <div
          className="cWcebC" />
        <div
          className="cWcebC" />
        <div
          className="cWcebC" />
        <div
          className="cWcebC" />
        <div
          className="cWcebC" />
      </div>
    </div>
    <p
      className="jLJUnn">
      This is a talk about devoxx
    </p>
    
  </div>
  <div
    className="col-xs-12 col-md-6 eOYzku">
    <div
      className="row start-xs">
      <h3
        className="jJURiy"
        id="notesHeader">
        My Notes
      </h3>
    </div>
    <textarea
      className="dBQlNn"
      placeholder="Write notes here..." />
    <div
      className="row">
      <h3
        className="jJURiy">
        Top Tracks
      </h3>
    </div>
    <div
      className="row">
      <ul
        className="gvIwUo">
        <li
          className="cLQKgI">
          track1
        </li>
        <li
          className="cLQKgI">
          track2
        </li>
      </ul>
    </div>
  </div>
</div>
`;
